docs/docs/data-fetching.md

```diff
@@ -0,0 +1,221 @@
+
+Alternately, if you want to source data yourself you can use APIs Gatsby provides. Source plugins take advantage of the [`sourceNodes` API](/docs/node-apis/#sourceNodes) and the [`createNode` action](/docs/actions/#createNode) provided by Gatsby to make your data queryable during the build process. If you want to source data yourself you can add a section of code like this using the `createNode` API to add a node to your data layer manually:
+
+```js:title=gatsby-node.js
+const fetch = require(`node-fetch`)
+
+exports.sourceNodes = async ({
+  actions: { createNode },
+  createContentDigest,
+}) => {
+  // get data from GitHub API at build time
+  const result = await fetch(`https://api.github.com/repos/gatsbyjs/gatsby`)
+  const resultData = await result.json()
+  // create node for build time data example in the docs
+  createNode({
+    nameWithOwner: resultData.full_name,
+    url: resultData.html_url,
+    // required fields
+    id: `example-build-time-data`,
+    parent: null,
+    children: [],
+    internal: {
+      type: `Example`,
+      contentDigest: createContentDigest(resultData),
+    },
+  })
+}
+```
+
+This node created manually could be retrieved with a query like this:
+
+```graphql
+query {
+  example {
+    nameWithOwner
+    url
+  }
+}
+```
+
+#### Writing a query to gather the static data needed for a page
+
+With the source plugin installed and added to your config, you can write a [static query](/docs/use-static-query/) that will retrieve the necessary data from Gatsby's data layer while building the site.
+
+```jsx:title=src/pages/index.js
+import React from "react"
+import { graphql, useStaticQuery } from "gatsby" // highlight-line
+
+const IndexPage = () => {
+  // highlight-start
+  const gatsbyRepoData = useStaticQuery(graphql`
+    query {
+      github {
+        repository(name: "gatsby", owner: "gatsbyjs") {
+          id
+          nameWithOwner
+          url
+        }
+      }
+    }
+  `)
+  // highlight-end
+
+  return (
+    <section>
+      <p>
+        Build Time Data: Gatsby repo{` `}
+        <a href={gatsbyRepoData.github.repository.url}>
+          {gatsbyRepoData.github.repository.nameWithOwner} // highlight-line
+        </a>
+      </p>
+    </section>
+  )
+}
+
+export default IndexPage
+```
+
+> This data is gathered at build time and written to a JSON file, the code is rewritten as the site is built to import the JSON file and set `gatsbyRepoData` equal to the contents of the JSON file instead of the call to `useStaticQuery`, you can read more about this process in the Gatsby internals section on [Normal vs Static Queries](/docs/static-vs-normal-queries/#replacing-queries-with-json-imports)
```

Comment:
This was a pretty long run-on sentence, so I broke it up.
```suggestion
> This data is gathered at build time and written to a JSON file. As the build continues, the code is rewritten behind the scenes to dynamically import the JSON file and set `gatsbyRepoData` equal to the contents of the JSON file instead of the call to `useStaticQuery`. You can read more about this process in the Gatsby internals section on [Normal vs Static Queries](/docs/static-vs-normal-queries/#replacing-queries-with-json-imports)
```